README.md

```diff
@@ -18,6 +18,15 @@ of the pipeline description, and more.
  * **Mailing List**: image-builder@redhat.com
  * **Changelog**: <https://github.com/osbuild/osbuild/releases>
 
+### Principles
+
+1. OSBuild stages are never broken, only deprecated. The same manifest should always produce the same output.
+2. OSBuild stages should be explicit whenever possible instead of e.g. relying on the state of the tree.
+3. OSBuild's pipelines are independent, so the tree is expected to be empty at the beginning of each.
+4. Manifests are expected to be machine-generated, so OSBuild has no convenience functions to support manually created manifests.
+5. OSBuild has confinement around the build environment, but it is not a security feature.
```

Comment:
(super nitpick) Maybe `OSBuild has confinement around the build environment, but it's not a security feature and does not protect against malicious inputs`  (might be too long?).  It's fine as is what I try to convey with this suggestion is that the confinement will not work against an attack but I guess it's kina implied already. Anyway, fine as is, maybe worth bringing malicious inputs in there :)

Reply:
Alternatively "OSBuild confines the build environment but this confinement is not a security boundary".

Reply:
"The build environment is confined against accidental misuse, but this should not be considered a security boundary."?

Reply:
Went with @bcl's suggestion